### PR TITLE
Reworked a bit

### PR DESCRIPTION
**Pull request by TealNerd**
All features now work, like, fully as far as I can tell. Wasn't able to reproduce the bug that caused it to crash before and changed the logic a bit.

Metadata packet for health actually had the index wrong before so that wouldn't have worked

Not sure what the potion stuff was trying to accomplish but it was actually just changing the potion duration being sent

It used to do something with armor color but it didn't seem necessary so I took that out. Also changed a bit how the enchant handling works and should be gucci now

### CODE REVIEW

File: src/com/civpvp/attrhider/AttrHider.java [hunk 1]
```diff
@@ -33,129 +26,84 @@
...
+					}
+					event.getPacket().getItemModifier().write(0, clean);
+				}
+			});
+		
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EFFECT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					if(event.getPacket().getIntegers().read(0) != event.getPlayer().getEntityId()) {
```
Comment (ProgrammerDan) [inline src/com/civpvp/attrhider/AttrHider.java:70]: Why did you remove `PacketContainer p = e.getPacket();`  ? That was the correct pattern to use, when you access a field multiple times.

File: src/com/civpvp/attrhider/AttrHider.java [hunk 2]
```diff
@@ -33,129 +26,84 @@
...
-	    	}
-	    });
-	    
-	    //Make reported health random
+	public void registerPacketAdapters() {
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EQUIPMENT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					ItemStack old = event.getPacket().getItemModifier().read(0);
```
Comment (ProgrammerDan) [inline src/com/civpvp/attrhider/AttrHider.java:41]: Similar here; retain the `StructureModifier<ItemStack> items` object as you reuse it at the end, no reason to go back up the stack when you can retain a local reference. Helps avoid secondary effects (for instance, getItemMeta() returns a clone,not the original object, etc. and the accessors you are using could have secondary effects of which you are unaware)

File: src/com/civpvp/attrhider/AttrHider.java [hunk 3]
```diff
@@ -33,129 +26,84 @@
...
-								watch.setValue(20f);
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_METADATA) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					Player player = event.getPlayer();
+					Entity entity = getEntityById(event.getPlayer().getWorld(), event.getPacket().getIntegers().read(0));
+					
+					if(entity != null && entity instanceof LivingEntity && entity.getPassenger() != player 
+							&& entity != player && !(entity instanceof Wither) && !(entity instanceof EnderDragon)) {
+						List<WrappedWatchableObject> list = event.getPacket().getWatchableCollectionModifier().read(0);
```
Comment (ProgrammerDan) [inline src/com/civpvp/attrhider/AttrHider.java:88]: Same here, retain the reference to the StructureModifier.

**Review by ProgrammerDan — CHANGES_REQUESTED** [review]

File: src/com/civpvp/attrhider/AttrHider.java [hunk 4]
```diff
@@ -33,129 +26,84 @@
...
-	    });
-	    
-	    //Make reported health random
+	public void registerPacketAdapters() {
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EQUIPMENT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					ItemStack old = event.getPacket().getItemModifier().read(0);
+					if(old == null) return;
```
Comment (Maxopoly) [inline src/com/civpvp/attrhider/AttrHider.java:42]: Just want to note that I absolutely hate this notation

File: src/com/civpvp/attrhider/AttrHider.java [hunk 5]
```diff
@@ -33,129 +26,84 @@
...
+					ItemStack clean = new ItemStack(old.getType());
```
Comment (Maxopoly) [inline src/com/civpvp/attrhider/AttrHider.java:43]: You forgot durability here, this will always reset durability of items sent to 0, which is bad for any block or item that has variations. Tools and armour should have their dura set to a fixed value though. That is the reason why the shouldBeObfuscated method exists and is still needed.

File: src/com/civpvp/attrhider/AttrHider.java [hunk 6]
```diff
@@ -33,129 +26,84 @@
...
-						event.setPacket(event.getPacket().deepClone());
-						StructureModifier<List<WrappedWatchableObject>> watcher = event.getPacket().getWatchableCollectionModifier();
-					  	for (WrappedWatchableObject watch : watcher.read(0)) {
-							if ((watch.getIndex() == 6) && (((Float)watch.getValue()).floatValue() > 0.0F)) {
-								watch.setValue(20f);
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_METADATA) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					Player player = event.getPlayer();
+					Entity entity = getEntityById(event.getPlayer().getWorld(), event.getPacket().getIntegers().read(0));
```
Comment (Maxopoly) [inline src/com/civpvp/attrhider/AttrHider.java:84]: This will iterate over a few thousand entities every time the health of any entity changes. Why not just do it the way it was done previously?

File: src/com/civpvp/attrhider/AttrHider.java [hunk 7]
```diff
@@ -33,129 +26,84 @@
...
+		for(Entity e : world.getEntities()) {
+			if(e.getEntityId() == id) {
+				return e;
 			}
-		});
+		}
+		return null;
 	}
-    
-    private boolean shouldBeObfuscated(Material type) {
```
Comment (Maxopoly) [inline src/com/civpvp/attrhider/AttrHider.java:122]: The functionality of this should stay, but making it a hashset would be a lot better

**Review by Maxopoly — CHANGES_REQUESTED** [review]
I dont think anything aside from the health id fix and the banner meta should be changed.

Or are there other issues with the things you changed that I'm not aware of?
